content-src/components/DiscoveryStreamComponents/TopSites/TopSites.jsx

```diff
@@ -4,14 +4,74 @@
 
 import { connect } from "react-redux";
 import { TopSites as OldTopSites } from "content-src/components/TopSites/TopSites";
+import { TOP_SITES_MAX_SITES_PER_ROW } from "common/Reducers.jsm";
 import React from "react";
 
 export class _TopSites extends React.PureComponent {
+  insertSpocContent(TopSites, data, promoAlignment) {
+    if (
+      !TopSites.rows ||
+      TopSites.rows.length === 0 ||
+      !data.spocs ||
+      data.spocs.length === 0
+    ) {
+      return null;
+    }
+
+    const topSites = [...TopSites.rows];
+    const [topSiteSpoc] = data.spocs;
+    const organicTopSiteExists = topSites.some(
+      topSite =>
+        topSite && topSite.url === topSiteSpoc.url && topSite.type !== "SPOC"
+    );
+
+    if (organicTopSiteExists) {
+      return null;
```

Comment:
instead of returning should we should next spoc if available. This will help avoid an extra filterPinned check in DiscoveryStreamFeed.jsm